regipy/plugins/ntuser/shellbags_ntuser.py

```diff
@@ -107,10 +123,85 @@ def _parse_shell_item_path_segment(self, shell_item):
             else:
                 path_segment = '{{{0:s}}}'.format(shell_item.shell_folder_identifier)
 
+        elif isinstance(shell_item, pyfwsi.users_property_view):
```

Comment:
There is a lot of repeating logic here, can you please move this to a common function?

Reply:
Done